Code snippets/src/main/java/data_access/InMemAddCourseDAO.java

```diff
@@ -4,23 +4,35 @@
 import main.java.use_case.courses.AddCourseDataAccessInterface;
 
 import java.util.ArrayList;
+import java.util.Map;
 
 public class InMemAddCourseDAO implements AddCourseDataAccessInterface {
 
-    private final ArrayList<String> courses = new ArrayList<>();
+    private final ArrayList<Course> courses = new ArrayList<>();
 
     @Override
     public Course getCourse(String courseId) {
         return null; // Test case using this inMem DAO does not use this method currently
     }
 
+
+
     @Override
     public void saveCourse(Course course) {
-        this.courses.add(course.getId());
+        this.courses.add(course);
     }
 
     @Override
     public boolean existsByID(String courseID) {
-        return courses.contains(courseID);
+        ArrayList<String> courseIDs = new ArrayList<>();
+        for (Course course: courses) {
+            courseIDs.add(course.getId());
+        }
+        return courseIDs.contains(courseID);
+    }
+
+    @Override
```

Comment:
are theres overrides necessary since interfaces have abstract methods that need to be implemented?